internal/faults/common_test.go

```diff
@@ -0,0 +1,415 @@
+package faults
+
+import (
+	"strings"
+	"testing"
+	"time"
+)
+
+const (
+	defaultAddress = "testService.testNamespace.svc.cluster.local:12345"
+	method         = "testMethod"
+	minAbortCode   = 0
+	maxAbortCode   = 10
+)
+
+func TestGetCanonicalAddress(t *testing.T) {
+	testCases := []struct {
+		name    string
+		address string
+		want    string
+	}{
+		{
+			name:    "cluster local address",
+			address: "testService.testNamespace.svc.cluster.local:12345",
+			want:    "testService.testNamespace",
+		},
+		{
+			name:    "external address",
+			address: "foo.bar:12345",
+			want:    "foo.bar:12345",
+		},
```

Comment:
this does not sound right to me. in one of the case we removed the `.svc.cluster.local` suffix and removed the port from the address; in the other case we kept the port (because it does not have `.svc.cluster.local` suffix).

so this function is inconsistent regarding whether it removes the port. this goes back to my question: https://github.com/reddit/baseplate.go/pull/666#discussion_r1874117167

Reply:
That's fair. After thinking about it more, if we are simulating a server being down entirely anyways, then stripping the port on external addresses should be done as well since we don't necessarily want to treat different ports differently. Updated to strip on all addresses. PTAL.